server/src/middleware.ts

```diff
@@ -39,3 +41,28 @@ export function versionedMiddleware(
   req.versioned = new VersionedMethods(req);
   next();
 }
+
+/**
+ * A JSON body parsing middleware that also records the raw size of the request
+ * body as `request.bodyByteLength`.
+ */
+export function parseJsonBody(
+  options: bodyParser.OptionsJson
+): ReturnType<typeof bodyParser.json> {
+  const { verify, ...otherOptions } = options;
+
+  return bodyParser.json({
+    ...otherOptions,
+    verify(
+      request: AppRequest,
+      response: Response,
+      buffer: Buffer,
+      encoding: string
+    ) {
+      request.bodyByteLength = buffer.byteLength || buffer.length;
+      if (verify) {
+        verify(request, response, buffer, encoding);
```

Comment:
It's a little confusing here that this function is named `verify` and then inside calls a function `verify that is not itself. It might be less confusing to just make the call here as `options.verify` instead to differentiate the two. Alternately, it might be clearer to writ line 56 as `verify: (...) => { ...`?